src/domain/project/project.resolver.ts

```diff
@@ -1,4 +1,81 @@
-import { Resolver } from '@nestjs/graphql';
-
+import { UseGuards } from '@nestjs/common';
+import { Args, Mutation, Query, Resolver } from '@nestjs/graphql';
+import { GqlAuthGuard } from '../../utils/authentication/graphql.guard';
+import { Roles } from '../../utils/decorators/roles.decorator';
+import { Profiling } from '../../utils/logging/logging.profiling.decorator';
+import { AspectInput } from '../aspect/aspect.dto';
+import { Aspect } from '../aspect/aspect.entity';
+import { IAspect } from '../aspect/aspect.interface';
+import { RestrictedGroupNames } from '../user-group/user-group.entity';
+import { ProjectInput } from './project.dto';
+import { Project } from './project.entity';
+import { IProject } from './project.interface';
+import { ProjectService } from './project.service';
 @Resolver()
-export class ProjectResolver {}
+export class ProjectResolver {
+  constructor(private projectService: ProjectService) {}
+
+  @Roles(
+    RestrictedGroupNames.CommunityAdmins,
+    RestrictedGroupNames.EcoverseAdmins
+  )
+  @UseGuards(GqlAuthGuard)
+  @Query(() => Project, {
+    nullable: false,
+    description: 'A particular Project, identified by the ID',
+  })
+  @Profiling.api
+  async opportunity(@Args('ID') id: number): Promise<IProject> {
```

Comment:
the method is named opportunity. is that going to impact the endpoint?

Reply:
Fixed...